src/v/archival/scrubber.cc

```diff
@@ -71,33 +74,196 @@ ss::future<scrubber::purge_result> scrubber::purge_partition(
           archival_log.error,
           "Remote delete disabled in tombstone on {}, refusing to purge",
           ntp);
-        co_return result;
+        co_return purge_result{.status = purge_status::success, .ops = 0};
```

Comment:
The lifecycle markers in the topic table are deleted after the purge is complete, so that shouldn't be possible. I'm not too sure what you mean though.